controllers/ProposalController.php

```diff
@@ -5,19 +5,47 @@
 
 
 use app\controllers\mainController\MainController;
+use app\models\databaseModels\Comment;
 use app\models\databaseModels\Proposal;
+use app\models\databaseModels\ProposalContentHistory;
+use app\models\databaseModels\Review;
 use yii\data\ActiveDataProvider;
+use yii\db\Exception;
 use yii\db\Query;
+use yii\web\NotFoundHttpException;
 
 class ProposalController extends MainController
 {
     /**
-     * Returns all proposals submitted by a member
+     * Returns all proposals submitted by a member or a single proposal if
+     * an Id was specified in url.
      *
+     * @param null|int $id
      * @return string
+     * @throws Exception
      */
-    public function actionMyProposals(): string
+    public function actionMyProposals(int $id = null): string
     {
+        if (!is_null($id)) {
+            $selectedProposal = $this->checkIfProposalExists($id);
+            $this->checkIfUserIsOwnerOfProposal($selectedProposal->submitter->id);
+            $chronologicalStream = $this->generateChronologicalStream
+            (
+                $selectedProposal->comments,
+                $selectedProposal->reviews,
+                $selectedProposal->proposalContentHistories
+            );
+            $chronologicalStream = $this->sortChronologicalStreamByDate($chronologicalStream);
```

Comment:
Why not just generate a sorted chronological stream with the generateChronologicalStream function ?

Reply:
You're right, a not chronological stream is never needed